spec/multi_patient_api/bulk_data_group_export_validation_spec.rb

```diff
@@ -0,0 +1,514 @@
+require_relative '../../lib/multi_patient_api/bulk_data_group_export_validation'
+require 'NDJSON'
+
+RSpec.describe MultiPatientAPI::BulkDataGroupExportValidation do
+  let(:group) { Inferno::Repositories::TestGroups.new.find('bulk_data_group_export_validation') }
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:test_session) { repo_create(:test_session, test_group_id: 'bulk_data_group_export_validation') }
+  let(:endpoint) { 'https://www.example.com' }
+  let(:status_output) { "[{\"url\":\"#{endpoint}\"}]" }
+  let(:bearer_token) { 'token' }
+  let(:headers) { { 'Content-Type' => 'application/fhir+ndjson' } }
+  let(:contents) { String.new }
+  let(:contents_missing_element) { String.new }
+  let(:scratch) { {} }
+  let(:input) do
+    { requires_access_token: true,
+      status_output: status_output,
+      bearer_token: bearer_token }
+  end
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(
+        test_session_id: test_session.id,
+        name: name,
+        value: value,
+        type: runnable.config.input_type(name)
+      )
+    end
+
+    Inferno::TestRunner.new(test_session: test_session, test_run: test_run).run(runnable, scratch)
+  end
+
+  # TODO: Create after implementing TLS Tester Class.
+  describe 'tls endpoint test' do
+  end
+
+  describe '[NDJSON download requires access token] test' do
+    let(:runnable) { group.tests[1] }
+
+    it 'skips when status_ouput is not provided' do
+      result = run(runnable)
+
+      expect(result.result).to eq('skip')
+      expect(result.result_message).to eq('Could not verify this functionality when Bulk Status Output is not provided')
+    end
+
+    it 'skips when requires_access_token is not provided' do
+      result = run(runnable, { status_output: status_output })
+
+      expect(result.result).to eq('skip')
+      expect(result.result_message).to eq('Could not verify this functionality when requiresAccessToken is not provided')
+    end
+
+    # TODO: Figure out why this behaves weirdly. Suspicion: false is stored serially, and then interpreted as a truthy value.
```

Comment:
Does this need to be figured out? Seems like a problem.

Reply:
Was passing in bool when it just wanted a string